src/Umbraco.Web/Cache/DistributedCacheBinder.cs

```diff
@@ -80,5 +84,49 @@ public void HandleEvents(IEnumerable<IEventDefinition> events)
                 }
             }
         }
+
+        // Internal for tests
+        internal static IEnumerable<IEventDefinition> GetReducedEventList(IEnumerable<IEventDefinition> events)
+        {
+            var reducedEvents = new List<IEventDefinition>();
+
+            var gotDoumentType = false;
+            var gotMediaType = false;
+            var gotMemberType = false;
+
+            foreach (var evt in events)
+            {
+                if (evt.Sender.ToString().Contains(nameof(Core.Services.Implement.ContentTypeService)))
+                {
+                    if (gotDoumentType == false)
+                    {
+                        reducedEvents.Add(evt);
+                        gotDoumentType = true;
+                    }
+                }
+                else if (evt.Sender.ToString().Contains(nameof(Core.Services.Implement.MediaTypeService)))
+                {
+                    if (gotMediaType == false)
+                    {
+                        reducedEvents.Add(evt);
+                        gotMediaType = true;
+                    }
+                }
+                else if (evt.Sender.ToString().Contains(nameof(Core.Services.Implement.MemberTypeService)))
+                {
+                    if (gotMemberType == false)
+                    {
+                        reducedEvents.Add(evt);
+                        gotMemberType = true;
+                    }
+                }
+                else
+                {
+                    reducedEvents.Add(evt);
+                }
+            }
+
+            return reducedEvents;
```

Comment:
Doesn't this mean it will just return an empty list of events if none of the events are of the types being looked for in this method? Does that mean that we've essentially removed all events from occurring for non content type changes?

Reply:
That's not my intention... if it's a non content type change it should hit the `else` clause and be added to the list.  So only the content/media/member type based events are reduced down to at most one of each.

Reply:
Ah, yes i see that now 👍